evm/src/executor/inspector/cheatcodes/ext.rs

```diff
@@ -160,13 +160,43 @@ fn get_env(key: &str, r#type: ParamType, delim: Option<&str>) -> Result<Bytes, B
         .map_err(|e| e.into())
 }
 
-fn read_file(path: &str) -> Result<Bytes, Bytes> {
+macro_rules! check_path_allowed {
+    ($state:ident, $path:ident) => {
+        // Get parent directory of the file that user wants to read/write.
+        let directory = Path::new($path)
+            .parent()
+            .ok_or("Can't get parent directory for the provided path".to_string().encode())?;
+
+        // Check it's a directory and it exists.
+        if !directory.is_dir() {
+            return Err("Parent directory of the file is not a directory. Is it exist?"
+                .to_string()
+                .encode()
+                .into())
+        }
```

Comment:
this is too restrictive for paths like `./output.csv`, right?

Reply:
I think we won't need it with `root.join` approach, just prefix check of the resulting path should be sufficient